packages/lesswrong/components/thinkPage/ThinkPostPage.tsx

```diff
@@ -0,0 +1,50 @@
+// TODO: Import component in components.ts
+import React, { useEffect } from 'react';
+import { registerComponent, Components } from '../../lib/vulcan-lib';
+import { useSingle } from '@/lib/crud/withSingle';
+import { useLocation } from '@/lib/routeUtil';
+import { useRecordPostView } from '../hooks/useRecordPostView';
+
+export const ThinkPostPage = () => {
+
+  const { Error404, Loading, ThinkPost, ThinkWrapper } = Components;
+
+  const { params: {postId, sequenceId} } = useLocation();
+
+  const { document: post, loading } = useSingle({
+    documentId: postId,
+    collectionName: "Posts",
+    fragmentName: "PostsPage",
+    skip: !postId
+  });
+
+  const { document: sequence, loading: sequenceLoading } = useSingle({
+    documentId: sequenceId,
+    collectionName: "Sequences",
+    fragmentName: "SequencesPageWithChaptersFragment",
+    skip: !sequenceId
+  });
+
+  const { recordPostView } = useRecordPostView(post as PostsListBase);
```

Comment:
this should not need a type cast.  if the problem is that the post is sometimes undefined because e.g. the `useSingle` is still loading, that means this should live inside of `ThinkPost`